Topic1/src/com/topic1/MailList.java

```diff
@@ -0,0 +1,38 @@
+package com.topic1;
+
+import java.util.HashSet;
+import java.util.Iterator;
+
+public class MailList implements Subject {
```

Comment:
[High] The maillist is the observer.
The shoppingCart and products are the observable objects. When price is change, the observer is notified. 
Please review *Observer* pattern.
Now you are not notifying about these events